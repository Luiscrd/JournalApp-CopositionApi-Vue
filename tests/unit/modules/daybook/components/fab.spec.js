import { shallowMount } from '@vue/test-utils'
import Fab from '@/modules/daybook/components/Fab.vue'

describe('Pruebas en el Fab component', () => {

  test('Debe renderizar el componente correctamente', () => {

      const wrapper = shallowMount(Fab)

      expect( wrapper.html() ).toMatchSnapshot()

  });

  test('Debe mostrar el icono por defecto', () => {

    const wrapper = shallowMount(Fab)

    const ico = wrapper.find('i')

    expect(ico.classes('fa-plus')).toBeTruthy()

});

test('Debe mostrar el icono por argumento: fa-circle', () => {

    const wrapper = shallowMount(Fab, {
        props: {
            icon: 'fa-circle'
        }
    })

    const ico = wrapper.find('i')

    expect(ico.classes('fa-circle')).toBeTruthy()

});

test('Debe emitir el evento on:click cuando se hace click', () => {

    const wrapper = shallowMount(Fab)

    wrapper.find('button').trigger('click')

    expect(wrapper.emitted('on:click')).toBeTruthy()

});

})