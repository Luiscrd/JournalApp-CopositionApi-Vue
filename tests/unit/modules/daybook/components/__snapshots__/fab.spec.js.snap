// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Fab component Debe renderizar el componente correctamente 1`] = `<button class="btn btn-primary"><i class="fa fa-2x fa-plus"></i></button>`;
